test/Accum.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+module Accum
+( tests
+) where
+
+import qualified Control.Carrier.Accum.Church as C.Accum.Church
+import qualified Control.Carrier.Accum.Strict as C.Accum.Strict
+import           Control.Effect.Accum
+import qualified Control.Monad.Trans.Accum as T.Accum
+import           Data.Tuple (swap)
+import           Gen
+import qualified Monad
+import qualified MonadFix
+import           Test.Tasty
+import           Test.Tasty.Hedgehog
+
+tests :: TestTree
+tests = testGroup "Accum"
+  [ testGroup "AccumC (Church)" $
+    [ testMonad
+    -- , testMonadFix FIXME There is a bug here
```

Comment:
Might be worth double-checking the size of the trees you’re generating if this is an issue. I don’t immediately notice a smoking gun in `gen0`, but it is regrettably easy to define not-guaranteed-convergent generators for trees when doing property-based testing.

Reply:
**Edit:** Ignore this 😊 The function passed to `mfix` is strictly evaluating its argument.